stdlib/_ssl.pyi

```diff
@@ -0,0 +1,344 @@
+import sys
+from _typeshed import Incomplete, ReadableBuffer, StrOrBytesPath
+from collections.abc import Callable
+from ssl import (
+    SSLCertVerificationError as SSLCertVerificationError,
+    SSLContext,
+    SSLEOFError as SSLEOFError,
+    SSLError as SSLError,
+    SSLObject,
+    SSLSocket,
+    SSLSyscallError as SSLSyscallError,
+    SSLWantReadError as SSLWantReadError,
+    SSLWantWriteError as SSLWantWriteError,
+    SSLZeroReturnError as SSLZeroReturnError,
+)
+from typing import Any, overload
+from typing_extensions import Literal, NotRequired, Self, TypeAlias, TypedDict, final
+
+_PasswordType: TypeAlias = Callable[[], str | bytes | bytearray] | str | bytes | bytearray
+_PCTRTT: TypeAlias = tuple[tuple[str, str], ...]
+_PCTRTTT: TypeAlias = tuple[_PCTRTT, ...]
+_PeerCertRetDictType: TypeAlias = dict[str, str | _PCTRTTT | _PCTRTT]
+
+class _Cipher(TypedDict):
+    aead: bool
+    alg_bits: int
+    auth: str
+    description: str
+    digest: str | None
+    id: int
+    kea: str
+    name: str
+    protocol: str
+    strength_bits: int
+    symmetric: str
+
+class _CertInfo(TypedDict):
+    subject: tuple[tuple[tuple[str, str], ...], ...]
+    issuer: tuple[tuple[tuple[str, str], ...], ...]
+    version: int
+    serialNumber: str
+    notBefore: str
+    notAfter: str
+    subjectAltName: NotRequired[tuple[tuple[str, str], ...] | None]
+    OCSP: NotRequired[tuple[str, ...] | None]
+    caIssuers: NotRequired[tuple[str, ...] | None]
+    crlDistributionPoints: NotRequired[tuple[str, ...] | None]
+
+class _EmptyCertInfo(TypedDict): ...
+
+def _test_decode_cert(__path: str) -> _CertInfo: ...
```

Comment:
In general, we prefer not to add private functions/attributes/classes unless somebody specifically needs them for something. This helps reduce the maintenance burden for us (since these are usually implementation details that often change in patch releases), and means that IDEs are less likely to have their autocomplete suggestions cluttered with these usually-undocumented internals. If a private class is used as a base class for a public class, then we've obviously got no choice except to include the private class in the stub -- but it looks like there's a few instances in this PR (like this function here) where you're adding private things that _maybe_ aren't really necessary :)

Reply:
That makes sense; I'll do some cleaning up.